js/coinflex.js

```diff
@@ -0,0 +1,2390 @@
+        //     }
+        //
+        const currencyId = this.safeString (transfer, 'asset');
+        const timestamp = this.safeString (transfer, 'transferredAt');
+        const fromAccount = this.safeString (transfer, 'fromAccount');
+        const toAccount = this.safeString (transfer, 'toAccount');
+        let status = this.parseTransactionStatus (this.safeString (transfer, 'status'));
+        // if case of 'transfer', as status field is not available, 'transferredAt' prop should mean that transfer was done
+        if (status === undefined && timestamp !== undefined) {
+            status = 'ok';
+        }
+        return {
+            'id': this.safeString (transfer, 'id'),
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'currency': this.safeCurrencyCode (currencyId, currency),
+            'amount': this.safeNumber (transfer, 'quantity'),
+            'fromAccount': fromAccount,
+            'toAccount': toAccount,
+            'status': status,
+            'info': transfer,
+        };
+    }
+
+    async buildOrderRequest (market, type, side, amount, price = undefined, params = {}) {
+        const clientOrderId = this.safeString (params, 'clientOrderId');
+        const maxCOI = '9223372036854775807';
+        if ((clientOrderId !== undefined) && Precise.stringGt (clientOrderId, maxCOI)) {
+            throw new InvalidOrder (this.id + ' createOrder() param clientOrderId should not exceed ' + maxCOI);
+        }
+        const orderType = this.convertOrderType (type);
+        const stopPrice = this.safeNumber (params, 'stopPrice');
+        const limitPrice = this.safeNumber (params, 'limitPrice');
+        const stopPriceIsDefined = stopPrice !== undefined;
+        const orderTypeIsStop = orderType === 'STOP';
+        if (orderTypeIsStop) {
+            if (!stopPriceIsDefined) {
+                throw new ArgumentsRequired (this.id + ' createOrder() requires params["stopPrice"] for stop orders');
+            } else if (limitPrice === undefined && price === undefined) {
+                throw new ArgumentsRequired (this.id + ' createOrder() requires "price" argument or params["limitPrice"] as a limit price for stop orders, as stop-market orders are not supported on this exchange');
+            }
+        }
+        await this.loadMarkets ();
+        const order = {
+            'marketCode': market['id'],
+            'side': this.convertOrderSide (side),
+            'orderType': this.convertOrderType (type),
+            'quantity': this.amountToPrecision (market['symbol'], amount),
+        };
+        if (stopPriceIsDefined || orderTypeIsStop) {
+            order['stopPrice'] = this.priceToPrecision (market['symbol'], stopPrice);
+            params = this.omit (params, 'stopPrice');
+            order['orderType'] = 'STOP';
+        }
+        // stop orders have separate field for limit price, so make further checks
+        if (type === 'limit') {
```

Comment:
I believe we're missing `|| orderTypeIsStop` otherwise orders with "stop" type won't get the required `limitPrice`

Reply:
correct! missed to update that too after I accepted STOP in types.

Reply:
updated, lmk if current logic seems ok.

Reply:
@ttodua seems ok now 🙂